packages/shared/src/utils/helpers/barrier.ts

```diff
@@ -0,0 +1,28 @@
+type TContract = {
+    high_barrier?: null | string;
+    barriers?: number;
+    barrier?: null | string;
+    low_barrier?: null | string;
+    expiry_type: string;
+};
+
+export const buildBarriersConfig = (contract: TContract, barriers = { count: contract.barriers }) => {
+    if (!contract.barriers) {
+        return undefined;
+    }
+
+    const obj_barrier = {};
+
+    ['barrier', 'low_barrier', 'high_barrier'].forEach(field => {
+        if (field in contract) obj_barrier[field] = contract[field as keyof typeof contract];
```

Comment:
when I added the type of **obj_barrier[field]** like this `obj_barrier[field as keyof typeof obj_barrier]`
I get this error:
**Type 'string | number | null | undefined' is not assignable to type 'never'.
  Type 'undefined' is not assignable to type 'never'.ts(2322)**

Reply:
```
type TObjectBarrier = Pick<TContract, 'barrier' | 'low_barrier' | 'high_barrier'>;
```

```suggestion
        const obj_barrier: TObjectBarrier = {};

    ['barrier', 'low_barrier', 'high_barrier'].forEach(field => {
        if (field in contract) obj_barrier[field as keyof TObjectBarrier] = contract[field as keyof TObjectBarrier];
```